File: Lista3/Exec3/src/exec3/Retangulo.java
import java.lang.Math.*;

/**
 * Classe retangulo
 */
public class Retangulo extends Forma {

    /**
     * Retangulo com 2 ladosiguais e 2 diferentes
     * @param lado1
     * @param lado2
     * @param cheio
     * @param cor
     */
    public Retangulo(double lado1, double lado2, boolean cheio, String cor){
        super(lado1, lado2, cheio, cor);
    }

    /**
     * tostring
     */
    @Override
    public String toString(){
        boolean cheio = getCheio();
        String s = "\nForma Geométrica: RETANGULO\n";
        s += "Lados: " + getLados()[0] + " e " + getLados()[1] + "\n";
        s += "Area: " + area() + "\n";
        s += "Perímetro: " + perimetro() + "\n";
        s += "Colorido: " + cheio + "\n";
        if(cheio){ 
            s += "Cor: " + getCor() + "\n"; 
        }
        return s;
    }

}

File: Lista4/exec3/retangulo.py
from forma import *

class retangulo(forma):

    def __init__(self, lado1: float, lado2: float, cheio: bool, cor: str) -> None:
        super().__init__(lado1, lado2, cheio, cor)

    def __str__(self) -> str:
        self.cheio = self.getCheio()
        s = '\nForma Geométrica: RETANGULO\n'
        s += "Lados: " + str(self.getLados()[0]) + " e " + str(self.getLados()[1]) + "\n"
        s += "Area: " + str(self.area()) + "\n"
        s += "Perímetro: " + str(self.perimetro()) + "\n"
        s += "Colorido: " + str(self.cheio) + "\n"
        if(self.cheio):
            s += "Cor: " + self.getCor() + "\n"
        
        return s

